cmd/fleet/serve.go

```diff
@@ -163,6 +163,22 @@ the way that the Fleet server works.
 				os.Exit(1)
 			}
 
+			// Check to see if the flag is populated
+			// Check if file exists on disk
+			// If file exists read contents
+			if config.Auth.JwtKeyPath != "" {
+				if _, err := os.Stat(config.Auth.JwtKeyPath); err == nil {
+					fileContents, err := ioutil.ReadFile(config.Auth.JwtKeyPath)
+					if err == nil {
+						config.Auth.JwtKey = strings.TrimSpace(string(fileContents))
+					} else {
+						initFatal(err, "Could not read the JWT Key file provided")
+					}
+				} else {
+					initFatal(err, "JWT Key file path provided does not exist")
+				}
```

Comment:
In this codebase we generally try to do the error handling first to reduce levels of nesting. This would be written as

```suggestion
				if _, err := os.Stat(config.Auth.JwtKeyPath); err != nil {
					initFatal(err, "JWT Key file path provided does not exist")
				}
				fileContents, err := ioutil.ReadFile(config.Auth.JwtKeyPath)
				if err != nil {
					initFatal(err, "Could not read the JWT Key file provided")
				}
				
				config.Auth.JwtKey = strings.TrimSpace(string(fileContents))
```

I also don't think the `os.Stat` check is sufficient (could be statting a directory, for example). IMO you could actually just leave it out entirely and handle the error from `ioutil.ReadFile`:

```suggestion
				fileContents, err := ioutil.ReadFile(config.Auth.JwtKeyPath)
				if err != nil {
					initFatal(err, "Could not read the JWT Key file provided")
				}
				
				config.Auth.JwtKey = strings.TrimSpace(string(fileContents))
```